### PR TITLE
Rename ICanHasViewContext to IViewContextAware

### LINKED ISSUE #4073: Rename ICanHasViewContext to IViewContextAware
Design notes here: https://github.com/aspnet/Mvc/issues/3973#issuecomment-182476618

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #4073

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @rynowak \ @lodejard

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/IViewContextAware.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/IViewContextAware.cs]: This should move out of `.Internal` now that we're keeping it.

**Comment by rynowak** [comment]
:shipit: after changing namespace

**Comment by lodejard** [comment]
Goodbye, ICouldHadViewContext :D

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/IViewContextAware.cs (hunk 1 above)
Comment (Eilon) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/IViewContextAware.cs]: :+1:

**Pull request closed without merging**